File: app/src/main/java/de/bitb/astroskop/ui/base/BasePresenter.java
package de.bitb.astroskop.ui.base;

import android.content.Context;

import java.lang.ref.WeakReference;

import de.bitb.astroskop.AstroApplication;
import de.bitb.astroskop.injection.IInjection;

public class BasePresenter<T extends IBaseView> {

    private WeakReference<T> view;

    public BasePresenter(AstroApplication application, T view) {
        this.view = new WeakReference<>(view);
        if (this instanceof IInjection) {
            ((IInjection) this).inject(application.getAppComponent());
        }
    }

    public T getView() {
        return view.get();
    }

    public Context getContext() {
        return getView().getContext();
    }

}

File: app/src/main/java/de/bitb/astroskop/ui/base/BasePresenter.kt
package de.bitb.astroskop.ui.base

import android.content.Context

import java.lang.ref.WeakReference

import de.bitb.astroskop.BaseApplication
import de.bitb.astroskop.injection.IInjection

open class BasePresenter<T : IBaseView>(application: BaseApplication, view: T) {

    private val view: WeakReference<T>

    val context: Context
        get() = getView().context

    init {
        this.view = WeakReference(view)
        if (this is IInjection) {
            (this as IInjection).inject(application.appComponent)
        }
    }

    fun getView(): T {
        return view.get()
    }

}
